canals/testing/test_component.py

```diff
@@ -18,9 +21,8 @@ def assert_can_be_saved_and_loaded_in_pipeline(self, component, tmp_path):
         """
         pipe = Pipeline()
         pipe.add_component("component", component)
-        save_pipelines({"pipe": pipe}, tmp_path / "test_pipe.json")
-
-        loaded_pipes = load_pipelines(tmp_path / "test_pipe.json")
-        new_pipe = loaded_pipes["pipe"]
-
-        assert pipe == new_pipe
+        data = pipe.to_dict()
+        test_file = tmp_path / "test_pipe.json"
+        test_file.write_text(json.dumps(data))
+        data = json.loads(test_file.read_text())
+        assert pipe == Pipeline.from_dict(data)
```

Comment:
The only reason why I was testing in a pipeline is because there weren't method to serialize the components in isolation. Do you think using the pipeline still has any value? If so I'd rather make two tests. Otherwise serializing only the component should suffice.

Reply:
Good point, I think we can remove this completely. 
Components that reimplement `from_dict` and `to_dict` have that responsibility now.

I would do this in a later PR though. Don't want to make this even bigger.